shared_model/utils/visitor_apply_for_all.hpp

```diff
@@ -30,7 +30,14 @@ namespace shared_model {
     class ToStringVisitor : public boost::static_visitor<std::string> {
      public:
       template <typename InputType>
-      std::string operator()(const InputType &operand) const {
+      auto operator()(const InputType &operand) const
+          -> decltype(operand.toString(), std::string()) {
```

Comment:
That really hard to read and understand, what's that for? Inheritance from static_visitor is already provide the type.

Reply:
It does not help because we should do SFINAE for `InputType` which has `operator->`, and reference `InputType`.

Reply:
ok, clear